lib/rubocop/cop/rails/time_zone.rb

```diff
@@ -183,6 +185,8 @@ def method_send?(node)
 
         def safe_method(method_name, node)
           if %w[new current].include?(method_name)
+            return 'parse' if node.children[2]&.type == :str
```

Comment:
```suggestion
            return 'parse' if node.first_argument&.str_type?
```

Reply:
Furthermore, simplifying the logic as follows would likely be better, as it reduces redundancy.

```diff
diff --git a/lib/rubocop/cop/rails/time_zone.rb b/lib/rubocop/cop/rails/time_zone.rb
index 0dce0e353..698c9a242 100644
--- a/lib/rubocop/cop/rails/time_zone.rb
+++ b/lib/rubocop/cop/rails/time_zone.rb
@@ -97,13 +97,9 @@ module RuboCop
         end

         def autocorrect_time_new(node, corrector)
-          if node.children[2]&.type == :str
-            corrector.replace(node.loc.selector, 'parse')
-          elsif node.arguments?
-            corrector.replace(node.loc.selector, 'local')
-          else
-            corrector.replace(node.loc.selector, 'now')
-          end
+          replacement = replacement(node)
+
+          corrector.replace(node.loc.selector, replacement)
         end

         # remove redundant `.in_time_zone` from `Time.zone.now.in_time_zone`
@@ -185,14 +181,20 @@ module RuboCop

         def safe_method(method_name, node)
           if %w[new current].include?(method_name)
-            return 'parse' if node.children[2]&.type == :str
-
-            node.arguments? ? 'local' : 'now'
+            replacement(node)
           else
             method_name
           end
         end

+        def replacement(node)
+          if node.arguments?
+            node.first_argument.str_type? ? 'parse' : 'local'
+          else
+            'now'
+          end
+        end
+
         def check_localtime(node)
           selector_node = node
```